src/simplify_algebra.cpp

```diff
@@ -1628,6 +1628,35 @@ struct find_split_transpose
     }
 };
 
+struct find_pow2_div
+{
+    auto matcher() const
+    {
+        return match::name("div")(
+            match::arg(0)(match::name("pow")(match::arg(0)(match::any().bind("x")),
+                                             match::arg(1)(match::has_value(2.0f)))
+                              .bind("pow")),
+            match::arg(1)(match::is_constant().bind("n")));
+    }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        auto ins = r.result;
+        auto pow = r.instructions["pow"];
+        auto n   = r.instructions["n"];
+        auto x   = r.instructions["x"];
+
+        // sqrt(n) with integral numbers can cause huge accuracy loss due to float->int rounding
+        if(shape::is_integral(n->get_shape().type()))
```

Comment:
We should also be checking for low precision. We probably shouldn't do this rewrite with fp32.